packages/picasso/src/utils/use-body-scroll-lock.ts

```diff
@@ -0,0 +1,53 @@
+import { isBrowser } from '@toptal/picasso-shared'
+import { useEffect, useMemo } from 'react'
+
+const layers = new Set<number>();
+let scrollLock: { prevBodyOverflow: string } | undefined = undefined;
+
+export const useBodyScrollLock = (isLocked: boolean) => {
+  const layerId = useMemo(generateLayerId, [])
+
+  useEffect(() => {
+    if (isLocked) {
+      layers.add(layerId)
+      dropScrollLock()
+    }
+
+    return () => {
+      layers.delete(layerId)
+      if (layers.size === 0) {
+        liftScrollLock()
+      }
+    }
+  }, [layerId, isLocked])
+}
+
+const generateLayerId = (() => {
+  let count = 0
+
+  return () => ++count
+})()
+
+const dropScrollLock = () => {
+  if (!isBrowser()) {
+    return
+  }
+
+  if (!scrollLock) {
+    scrollLock = {
+      prevBodyOverflow: window.getComputedStyle(document.body).overflow,
+    }
+    document.body.style.overflow = 'hidden'
+  }
```

Comment:
I know that this was the previous behavior, but it doesn't feel right, `getComputedStyle()` would get the value from stylesheets which is not the true value for `style="overflow: ..."`, setting it back to `style.overflow` feels like a mistake, what do you think?

For example we could have this structure:
```html
<div class="foo">
  ...
</div>
<style>
.foo {
  overflow: hidden;
}
</style>
```
After scroll locking we would get `overflow: hidden` from the stylesheet and when unlocking we would set it back to the style attribute. The structure would become:
```html
<div class="foo" style="overflow: hidden">
  ...
</div>
<style>
.foo {
  overflow: hidden;
}
</style>
```
While this is rare problem it could cause race conditions when classes are being added or removed from `body`, or any dynamic stylesheets for that matter. I would suggest just using the prop itself, instead of `getComputedStyles`:
```suggestion
  if (!scrollLock) {
    scrollLock = {
      prevBodyOverflow: document.body.style.overflow,
    }
    document.body.style.overflow = 'hidden'
  }
```